arcgis-android-toolkit/src/main/java/com/esri/arcgisruntime/toolkit/floorfilter/FloorFilterView.kt

```diff
@@ -0,0 +1,1071 @@
+import android.widget.TextView
+import androidx.annotation.Dimension
+import androidx.appcompat.content.res.AppCompatResources
+import androidx.constraintlayout.widget.ConstraintLayout
+import androidx.constraintlayout.widget.ConstraintSet
+import androidx.core.widget.TextViewCompat
+import androidx.databinding.DataBindingUtil
+import androidx.databinding.ViewDataBinding
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.esri.arcgisruntime.mapping.GeoModel
+import com.esri.arcgisruntime.mapping.floor.FloorFacility
+import com.esri.arcgisruntime.mapping.floor.FloorLevel
+import com.esri.arcgisruntime.mapping.floor.FloorSite
+import com.esri.arcgisruntime.mapping.floor.FloorManager
+import com.esri.arcgisruntime.mapping.view.GeoView
+import com.esri.arcgisruntime.mapping.view.MapView
+import com.esri.arcgisruntime.mapping.view.SceneView
+import com.esri.arcgisruntime.toolkit.R
+import com.esri.arcgisruntime.toolkit.extension.dpToPixels
+import com.esri.arcgisruntime.toolkit.extension.pixelsToDp
+import com.esri.arcgisruntime.toolkit.extension.pixelsToSp
+
+class FloorFilterView: LinearLayout {
+
+    /**
+     * The siteId of the selected [FloorSite].
+     *
+     * @since 100.13.0
+     */
+    var selectedSiteId: String?
+        get() { return floorFilterManager.selectedSiteId }
+        set(value) { floorFilterManager.selectedSiteId = value }
+
+    /**
+     * The facilityId of the selected [FloorFacility].
+     *
+     * @since 100.13.0
+     */
+    var selectedFacilityId: String?
+        get() { return floorFilterManager.selectedFacilityId }
+        set(value) { floorFilterManager.selectedFacilityId = value }
+
+    /**
+     * The levelId of the selected [FloorLevel].
+     *
+     * @since 100.13.0
+     */
+    var selectedLevelId: String?
+        get() { return floorFilterManager.selectedLevelId }
+        set(value) { floorFilterManager.selectedLevelId = value }
+
+    /**
+     * The [FloorManager] of the attached [GeoModel].
+     *
+     * @since 100.13.0
+     */
+    val floorManager: FloorManager?
+        get() { return floorFilterManager.floorManager }
+
+    // Custom events
+
+    /**
+     * Called when the selected [FloorSite], [FloorFacility], or [FloorLevel] changes.
+     *
+     * @since 100.13.0
+     */
+    private var onSelectionChangeListener: (() -> Unit)? = null
+    fun setOnSelectionChangeListener(onSelectionChangeListener: (() -> Unit)?) {
+        this.onSelectionChangeListener = onSelectionChangeListener
+    }
+
+    // UI related parameters
+
+    /**
+     * The [Int] used to determine the height of the level, close, and site/facility buttons
+     * in the [floorsRecyclerView].
+     *
+     * The default is 40dp.
+     * Use [setButtonSize] to change the height of the buttons.
+     *
+     * @since 100.13.0
+     */
+    private var buttonHeightDp: Int
+        set(value) { uiParameters.buttonHeightDp = value }
```

Comment:
Please update the formatting of the getter/setter properties. I am not going to comment anywhere else.